doc/nrf/releases_and_maturity/migration/migration_3.1_54h_suit_ironside.rst

```diff
@@ -0,0 +1,173 @@
+-----------------------
+
+You must update your devicetree files as follows:
+
+1. Remove the old UICR partition.
+   In your board's DTS overlay, remove any node that defined the ``uicr`` partition.
+
+#. Add the PERIPHCONF array.
+   In your devicetree, under the ``mram1x`` partitions node, define a partition node labeled ``peripconf_partition`` with a size of at least 8 KB to embed the generated address-value blob.
+
+#. Remove IPC-shared-memory reservation.
+   As IronSide relocates the IPC buffer to a fixed RAM20 address, you can delete any manual reservation in RAM0.
+   Refer to the `Memory map changes`_ section.
+
+#. Update IPC configuration for IronSide SE.
+   The shared memory for communication with the Secure Domain now uses fixed addresses in ``RAM20``.
+   A single memory region is used for both RX and TX operations.
+   The IPC nodes use the ``nordic,ironside-call`` compatible and communicate using the new *IronSide Calls* IPC driver.
+
+   For custom board devicetree files, you can copy the IPC configuration from the nRF54H20 DK reference implementation.
+   The devicetree defines the shared memory region and IPC nodes as follows:
+
+   .. code-block:: dts
+
+      // Shared memory region in RAM20
+      cpusec_cpuapp_ipc: memory@2f88f000 {
+          reg = <0x2f88f000 DT_SIZE_K(4)>;
+      };
+
+   .. code-block:: dts
+
+      // IPC nodes using IronSide calls driver
+      cpusec_cpuapp_ipc_tx: ipc@deadbeef {
+          compatible = "nordic,ironside-call";
+          mboxes = <&cpuapp_cpusec_ipc 0>, <&cpuapp_cpusec_ipc 1>;
+          mbox-names = "rx", "tx";
+          memory-region = <&cpusec_cpuapp_ipc>;
+          status = "okay";
+      };
+
+   .. code-block:: dts
+
+      cpusec_cpuapp_ipc_rx: ipc@deadbeef {
+          compatible = "nordic,ironside-call";
+          mboxes = <&cpusec_cpuapp_ipc 2>, <&cpuapp_cpusec_ipc 3>;
+          mbox-names = "rx", "tx";
+          memory-region = <&cpusec_cpuapp_ipc>;
+          status = "okay";
+      };
+
+Update PERIPHCONF
+-----------------
+
+The new UICR format no longer holds peripheral configuration initial values.
+You must generate a PERIPHCONF blob at build time.
+
+The Zephyr build invokes the :file:`gen_uicr.py` script (:file:`soc/nordic/common/uicr/gen_uicr.py` in the Zephyr tree) via ``nrf-regtool`` in the NCS sysbuild system.
+When the following Kconfig options are set:
+
+  * :kconfig:option:`CONFIG_NRF_HALTIUM_GENERATE_UICR` to ``y``
+  * :kconfig:option:`CONFIG_NRF_HALTIUM_UICR_PERIPHCONF` to ``y``
+
+The script does the following:
+
+  1. It reads the ``peripconf_partition`` node in the devicetree to discover the partition's address and size.
+  #. It extracts the address/value pairs from the ``PERIPHCONF`` section of the Zephyr ELF image.
+  #. It generates two Intel HEX files:
+
+    * :file:`uicr.hex` - the new UICR entries
+    * :file:`periphconf.hex` - the MRAM-resident ``PERIPHCONF`` blob
+
+Both HEX files must be flashed alongside your firmware image.
```

Comment:
```suggestion
Both HEX files must be programmed alongside your firmware image.
```
https://nordicsemi.atlassian.net/wiki/spaces/TECHDOC/pages/120298016/Terms+abbreviations+and+spelling+rules#Terms,abbreviations,andspellingrules-P

(although it's not very clear compared to the entry for flash)